constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +499,56 @@ func TestRegoRewriterErrorCases(t *testing.T) {
 		}
 	}
 }
+
+func TestRegoRewriterParseErrors(t *testing.T) {
+	testcases := []RegoRewriterTestcase{
+		{
+			name:      "fail to add base",
+			baseSrcs:  map[string]string{"invalid.rego": RegoSrc("foo", Body("something invalid"))},
+			wantError: true,
+		},
+		{
+			name:      "fail to add lib",
+			libSrcs:   map[string]string{"invalid.rego": RegoSrc("foo", Body("something invalid"))},
+			wantError: true,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, tc.Run)
```

Comment:
I know this is the pattern in the file, but it obfuscates the functionality under test. As the test reader, from reading this test I can't tell what function is being called that returns the error.

Compare to:
```
tcs := []struct{
  name string
  path string
  src string
  wantError error
} {{
  name: "fail to add entrypoint",
  src: "something invalid",
}, ...}

for _, tc := range tcs {
  t.Run(tc.name, func() {
    rr, err := New(...)
    
    gotErr := rr.AddEntryPoint(tc.path, tc.content)
    if !errors.Is(tc.wantError, err) {
      t.Fatalf("got AddEntryPoint() error = %v, want %v", gotErr, tc.wantError)
    }
  })
}
```

This is fully self-contained. I don't have to go around the file to figure out what was being called, and what failed.

With that, as the test reader I can immediately see what is being tested at the exported API level rather than what is hidden away in a tc.Run() command. As a test writer, I can finely craft failure modes specific to the function being tested rather than worrying about making something that could break the ten or so other things being called in RegoRewriterTestcase.Run().

While this might result in more verbose test code, it is better to have more verbose (but clear) test code than to try to DRY it.

Reply:
Keeping the original `TestRegoRewriter` that used the `tc.Run` function, but I rewrote the tests to `TestRegoRewriteEntryPoint` and `TestRegoRewriteLib`